pandas/src/skiplist.h

```diff
@@ -68,33 +69,38 @@ static PANDAS_INLINE int int_min(int a, int b) {
 
 static PANDAS_INLINE node_t *node_init(double value, int levels) {
   node_t *result;
-  result = (node_t*) calloc(1, sizeof(node_t));
-
-  result->value = value;
-  result->levels = levels;
-  result->is_nil = 0;
-  result->ref_count = 0;
-
-  result->next = (node_t**) malloc(levels * sizeof(node_t*));
-  result->width = (int*) malloc(levels * sizeof(int));
-
+  result = (node_t*) malloc(sizeof(node_t));
+
+  if (result) {
+      result->value = value;
+      result->levels = levels;
+      result->is_nil = 0;
+      result->ref_count = 0;
+      result->next = (node_t**) malloc(levels * sizeof(node_t*));
+      result->width = (int*) malloc(levels * sizeof(int));
+      if (!(result->next && result->width) && (levels != 0)) {
+        free(result->next);
+        free(result->width);
+        free(result);
+        return NULL;
+      }
+  }
   return result;
 }
 
 // do this ourselves
-
 static PANDAS_INLINE void node_incref(node_t *node) {
-  node->ref_count += 1;
+  ++(node->ref_count);
 }
 
 static PANDAS_INLINE void node_decref(node_t *node) {
-  node->ref_count -= 1;
+  --(node->ref_count);
 }
 
 static void node_destroy(node_t *node) {
   int i;
   if (node) {
-    if (node->ref_count == 1) {
+    if (node->ref_count <= 1) {
```

Comment:
I think it makes clean-up code cleaner.  I will write something to test for leaks a bit later.